docs/admin/config/private_network.mdx

```diff
@@ -0,0 +1,70 @@
+# Private network configuration
+When deploying self-hosted Sourcegraph instances in private networks with specific compliance and policy requirements, additional configuration may be required to ensure all networking features are functioning correctly. The reasons for applying the following configuration options depend on the specific functionality of the Sourcegraph service and the unique network and infrastructure requirements of the organization. The following Sourcegraph services typically initiate outbound connections to external services:
+- **sourcegraph-frontend**: The frontend service is responsible for connecting to external [auth providers](../auth), sending [telemetry data](../pings), testing code host connections, and connecting to [externally hosted](../external_services) Sourcegraph services
+- **gitserver**: Executes git commands against externally hosted [code hosts](../external_service)
+- **repo-updater**: Communicates with [code hosts](../external_service) APIs to coordinate the repository synchronize
+- **migrator**: Connects to Postgres instances (which may be [externally hosted](../external_services/postgres)) to process database migrations
+- **executor**: Sourcegraph [Executor](../executors) batch change or precise indexing jobs may need to connect to services hosted within an organization's private network
+
+## HTTP proxy configuration
+All Sourcegraph services respect the conventional `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY` environment variables for routing Sourcegraph client application HTTP traffic through a proxy server. The steps for configuring proxy environment variables will depend on your Sourcegraph deployment method.
+
+### Kubernetes Helm
+Add the proxy environment variables to your Sourcegraph Helm chart [override file](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/values.yaml):
+
+```yaml
+frontend|gitserver|repo-updater:
+  env:
+  - name: HTTP_PROXY 
+    value: http://proxy.example.com:8080
+  - name: HTTPS_PROXY
+    value: http://proxy.example.com:8080
+  - name: NO_PROXY
+    value:  "*.cluster.local,blobstore,codeinsights-db,codeintel-db,sourcegraph-frontend-internal,sourcegraph-frontend,github-proxy,gitserver,grafana,indexed-search-indexer,indexed-search,jaeger-query,pgsql,precise-code-intel-worker,prometheus,redis-cache,redis-store,repo-updater,searcher,symbols,syntect-server,worker-executors,worker" 
+```
+
+<Callout type="warning">Failure to configure `NO_PROXY` correctly can cause proxy configuration to interfere with local networking between internal Sourcegraph services</Callout>
+
+## Using private CA root certificates
+Some organizations maintain a private Certificate Authority (CA) for issuing certificates within their private network. When Sourcegraph connects to TLS encrypted service using a self-signed certificate that it does not trust, it will throw the `x509: certificate signed by unknown authority` error message in logs.
```

Comment:
minor nitpick: ", it will throw the ..."  ===> ", you will see `x509: certificate signed by unknown authority` error messages in the logs"